packages/relayer/indexer/indexer.go

```diff
@@ -0,0 +1,383 @@
+package indexer
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log/slog"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/cyberhorsey/errors"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
```

Comment:
Shall we import and reuse this package in this repo: https://github.com/taikoxyz/taiko-client/blob/main/pkg/rpc/ethclient.go ? not sure if this repo will also face those kind of timeout issues?

Reply:
OK, did the rest of the comments, will do this one after the weekend I think!